x/gamm/types/account.go

```diff
@@ -337,9 +337,9 @@ func (pa PoolAccount) PokeTokenWeights(blockTime time.Time) {
 		// 		 Ensure the new asset has some token sent with it.
 		pa.updateAllWeights(params.TargetPoolWeights)
 		// We've finished updating weights, so delete this parameter
+		// TODO: This line doesn't work, since this is a non-pointer receiever,
+		// and pa.PoolParams gets copied.
 		pa.PoolParams.SmoothWeightChangeParams = nil
```

Comment:
An optimization that would be nice is to set `pa.PoolParams.SmoothWeightChangeParams = nil` once its hit the target weight. Unfortunately all the PoolAccountInterface methods have a non-pointer receiver at the moment, so this change won't get applied. (Because PoolParams is copied, as its not a slice =/)

This probably will get resolved in later refactors to change up how we use the interface vs concrete struct.

Reply:
Should we at least set it to an empty struct for now?

`pa.PoolParams.SmoothWeightChangeParams = SmoothWeightChangeParams{}`

Reply:
We'd have to keep the same target pool weights (so that we hit this case in the check), so doesn't seem worth it.